### DISCUSSION TITLE
Using Rails validations while creating account

### DISCUSSION THREAD

**Discussion by deepj** [janko/rodauth-rails#83] [Q&A]
Hello,
is there any recommended way to run Rails validations while creating account. I'd like to use classical Rails validations for additional attributes in Account model.

Example model:

```ruby
class Account < ApplicationRecord
  include Rodauth::Rails.model

  validates :name, presence: true
  validates :surname, presence: true
end
```

When I try to create an account via JSON API, I'd like to validate these attributes. Unfortunately, I couldn't find out a way how to run them and in case of any validation errors to return these validation errors in the JSON response. Something like

```json
{
  "field_error": [
    "name",
    "is blank"
  ],
  "error": "There was an error logging in"
}
```

**Comment by janko** [comment]
Hi, you'll have to write custom validations for these parameters inside your Rodauth configuration, see [this guide](http://rodauth.jeremyevans.net/rdoc/files/doc/guides/registration_field_rdoc.html).
